### PR TITLE
Add support for EXISTS expression

### PR DESCRIPTION
**Pull request by mfussenegger**
## Summary of the changes / Why this improves CrateDB

Closes https://github.com/crate/crate/issues/12888

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: server/src/main/java/io/crate/execution/engine/FirstColumnConsumers.java [hunk 1]
```diff
@@ -21,51 +21,51 @@
...
+import io.crate.data.Row;
+import io.crate.expression.symbol.SelectSymbol.ResultType;
+
 /**
  * Collectors to retrieve either {@link AllValues} or a {@link SingleValue} of the first column of each row.
  */
 public class FirstColumnConsumers {
 
-    private static class AllValues implements Collector<Row, Collection<Object>, Object[]> {
+    private static class AllValues implements Collector<Row, List<Object>, List<Object>> {
```
Comment (mfussenegger) [inline server/src/main/java/io/crate/execution/engine/FirstColumnConsumers.java:42]: I'm not entirely sure why this wasn't a problem before. Arrays are supposed to be of type `List<?>`, not `Object[]`. My best guess is that IN(..) and = ANY(...)  expressions run it through `ArrayType.value(..)` which casts `Object[]` to `List<Object>`, but I didn't bother validating - I couldn't trigger a user facing bug with the previous implementation.

File: server/src/test/java/io/crate/integrationtests/CorrelatedSubqueryITest.java [hunk 2]
```diff
@@ -152,9 +152,15 @@ public void test_correlated_subquery_within_case_using_outer_column_in_where_cla
                 END AS base_type
             FROM
                 pg_catalog.pg_type t
+            LIMIT 4
             """;
         execute(stmt);
-        assertThat(TestingHelpers.printedTable(response.rows())).isEqualTo("");
+        assertThat(TestingHelpers.printedTable(response.rows())).isEqualTo(
```
Comment (mfussenegger) [inline server/src/test/java/io/crate/integrationtests/CorrelatedSubqueryITest.java:158]: This assertion was wrong and the collector wasn't implemented correctly. Re-using `FirstColumnConsumers` fixed this.

File: docs/general/dql/selects.rst [hunk 3]
```diff
@@ -627,6 +627,32 @@ This behaviour applies to:
...
+
+The result only depends on the amount of rows, not on the actual values. Because
+of that it is common practice to use ``SELECT 1 [...]`` within the subquery.
+
+An example:
+
+::
+
+    cr> SELECT mountain FROM sys.summits t
+    ...   WHERE exists (SELECT 1 FROM sys.summits WHERE mountain = t.mountain)
```
Comment (seut) [inline docs/general/dql/selects.rst:646]: Was the operator written in lowercase by intend?

```suggestion
    ...   WHERE EXISTS (SELECT 1 FROM sys.summits WHERE mountain = t.mountain)
```

**Review by seut — APPROVED** [review]
👍 just a very minor suggestion

File: server/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java [hunk 4]
```diff
@@ -779,4 +780,39 @@ public void test_non_recursive_with_query() {
...
+
+        stmt = """
+            SELECT
+                table_name
+            FROM
+                information_schema.tables t
+            WHERE
+                EXISTS
+                (
+                 SELECT 1
```
Comment (matriv) [inline server/src/test/java/io/crate/integrationtests/SubSelectIntegrationTest.java:807]: Can we also have a test that selects an actual column of the table? 
maybe here: `SELECT column_name`

**Review by matriv — APPROVED** [review]
Great! Left a test suggestion

**Pull request merged**